include/ensmallen_bits/cmaes/cmaes_impl.hpp

```diff
@@ -206,64 +178,43 @@ typename MatType::elem_type CMAES<SelectionPolicyType>::Optimize(
     // Update Step Size.
     if (iterate.n_rows > iterate.n_cols)
     {
-      ps[idx1] = (1 - cs) * ps[idx0] + std::sqrt(
-          cs * (2 - cs) * muEffective) * covLower.t() * step;
+      ps[idx1] = (1 - params.csigma) * ps[idx0] + std::sqrt(
+          params.csigma * (2 - params.csigma) * params.mu_eff) * covLower.t() * step;
     }
     else
     {
-      ps[idx1] = (1 - cs) * ps[idx0] + std::sqrt(
-          cs * (2 - cs) * muEffective) * step * covLower.t();
+      ps[idx1] = (1 - params.csigma) * ps[idx0] + std::sqrt(
+          params.csigma * (2 - params.csigma) * params.mu_eff) * step * covLower.t();
     }
 
     const ElemType psNorm = arma::norm(ps[idx1]);
-    sigma(idx1) = sigma(idx0) * std::exp(cs / ds * ( psNorm / enn - 1));
+    const size_t hs = (psNorm / sqrt(1 - std::pow(1 - params.csigma, 2 * i)) < params.hsigma) ? 1 : 0;
+    const double deltahs = (1 - hs) * params.cc * (2 - params.cc);
 
     // Update covariance matrix.
-    if ((psNorm / sqrt(1 - std::pow(1 - cs, 2 * i))) < h)
-    {
-      pc[idx1] = (1 - cc) * pc[idx0] + std::sqrt(cc * (2 - cc) *
-        muEffective) * step;
-
-      if (iterate.n_rows > iterate.n_cols)
-      {
-        C[idx1] = (1 - c1 - cmu) * C[idx0] + c1 *
-          (pc[idx1] * pc[idx1].t());
-      }
-      else
-      {
-        C[idx1] = (1 - c1 - cmu) * C[idx0] + c1 *
-          (pc[idx1].t() * pc[idx1]);
-      }
-    }
-    else
-    {
-      pc[idx1] = (1 - cc) * pc[idx0];
-
-      if (iterate.n_rows > iterate.n_cols)
-      {
-        C[idx1] = (1 - c1 - cmu) * C[idx0] + c1 * (pc[idx1] *
-            pc[idx1].t() + (cc * (2 - cc)) * C[idx0]);
-      }
-      else
-      {
-        C[idx1] = (1 - c1 - cmu) * C[idx0] + c1 *
-            (pc[idx1].t() * pc[idx1] + (cc * (2 - cc)) * C[idx0]);
-      }
-    }
-
+    sigma(idx1) = sigma(idx0) * std::exp(params.csigma / params.dsigma * ( psNorm / params.chi - 1));
+    pc[idx1] = (1 - params.cc) * pc[idx0] + hs * std::sqrt(params.cc * (2 - params.cc) * params.mu_eff) * step; 
+    C[idx1] = (1 + params.c1 * deltahs - params.c1 - params.cmu * arma::accu(params.weights)) * C[idx0];
+  
     if (iterate.n_rows > iterate.n_cols)
     {
-      for (size_t j = 0; j < mu; ++j)
+      C[idx1] = C[idx1] + params.c1 * (pc[idx1] * pc[idx1].t());
+      for (size_t j = 0; j < params.lambda; ++j)
       {
-        C[idx1] = C[idx1] + cmu * w(j) *
+        if(params.weights(j) < 0) params.weights(j) *= params.dim/std::pow(arma::norm(z[j]), 2);
+        if(params.weights(j) == 0) break;
+        C[idx1] = C[idx1] + params.cmu * params.weights(j) *
             pStep[idx(j)] * pStep[idx(j)].t();
```

Comment:
```suggestion
        if (params.weights(j) < 0) params.weights(j) *= params.dim / std::pow(arma::norm(z[j]), 2);
        if (params.weights(j) == 0) break;
        C[idx1] = C[idx1] + params.cmu * params.weights(j) *
            pStep[idx(j)] * pStep[idx(j)].t();
```

Add missing spaces between operations.